docs/reference/setup/restart-cluster.asciidoc

```diff
@@ -0,0 +1,147 @@
+[[restart-cluster]]
+== Restarting a cluster
+ 
+There might be situations when you want to perform a full-cluster restart which 
+means you shut down and restart all the nodes in the cluster. The following 
+description guides you through the process step by step.
+
+. *Disable shard allocation.*
++
+--
+include::{docdir}/upgrade/disable-shard-alloc.asciidoc[]
+--
+
+
+. *Stop indexing and perform a synced flush.*
++
+--
+Performing a <<indices-synced-flush-api, synced-flush>> speeds up shard
+recovery.
+
+include::{docdir}/upgrade/synced-flush.asciidoc[]
+--
+
+. *Temporarily stop the tasks associated with active {ml} jobs and {dfeeds}.* (Optional)
++
+--
+You have two options to handle {ml} jobs and {dfeeds} when you shut down a 
+cluster:
+
+* Temporarily halt the tasks associated with your {ml} jobs and {dfeeds} and
+prevent new jobs from opening by using the 
+<<ml-set-upgrade-mode,set upgrade mode API>>:
++
+[source,console]
+--------------------------------------------------
+POST _ml/set_upgrade_mode?enabled=true
+--------------------------------------------------
++
+When you disable upgrade mode, the jobs resume using the last model state that 
+was automatically saved. This option avoids the overhead of managing active jobs 
+during the shutdown and is faster than explicitly stopping {dfeeds} and closing 
+jobs.
+
+* {stack-ov}/stopping-ml.html[Stop all {dfeeds} and close all jobs]. This option
+saves the model state at the time of closure. When you reopen the jobs after the
+cluster restart, they use the exact same model. However, saving the latest model 
+state takes longer than using upgrade mode, especially if you have a lot of jobs 
+or jobs with large model states.
+--
+
+. *Shut down all nodes.*
++
+--
+include::{docdir}/upgrade/shut-down-node.asciidoc[]
+--
+
+. *Perform the changes that require to shut down the cluster.*
```

Comment:
The wording is a bit awkward here. I could see someone misinterpreting this as "Perform the changes required to shut down the cluster."

Maybe just simplify it to "Perform any needed changes." It might be a good idea to append a sentence indicating that the cluster is now shut down.